src/WebJobs.Extensions.DurableTask/OutOfProcMiddleware.cs

```diff
@@ -347,14 +357,125 @@ public async Task CallActivityAsync(DispatchMiddlewareContext dispatchContext, F
                     ResponseEvent = new TaskFailedEvent(
                         eventId: -1,
                         taskScheduledId: scheduledEvent.EventId,
-                        reason: $"Function {functionName} failed with an unhandled exception.",
-                        exceptionDetails),
+                        reason: $"Function '{functionName}' failed with an unhandled exception.",
+                        details: null,
+                        GetFailureDetails(result.Exception)),
                 };
             }
 
             // Send the result of the activity function to the DTFx dispatch pipeline.
             // This allows us to bypass the default, in-process execution and process the given results immediately.
             dispatchContext.SetProperty(activityResult);
         }
+
+#if NET6_0_OR_GREATER
+        private static FailureDetails GetFailureDetails(Exception e)
+        {
+            if (e.InnerException != null && e.InnerException.Message.StartsWith("Result:"))
+            {
+                Exception rpcException = e.InnerException;
+                if (TryGetRpcExceptionFields(rpcException.Message, out string? exception, out string? stackTrace))
+                {
+                    if (TrySplitExceptionTypeFromMessage(exception, out string? exceptionType, out string? exceptionMessage))
+                    {
+                        return new FailureDetails(exceptionType, exceptionMessage, stackTrace, innerFailure: null, isNonRetriable: false);
+                    }
+
+                    return new FailureDetails("(unknown)", exception, stackTrace, innerFailure: null, isNonRetriable: false);
```

Comment:
I noticed `innerFailure` is always `null` in here. Why is that? Do we not except inner failures in OOProc PLs? I guess it might be worthwhile to define what an innerFailure would be in this case. I've been thinking of them as exceptions that were re-thrown, and therefore became wrapped in an outer exception.

Reply:
The problem is that Azure Functions doesn't preserve any of the structure of an exception or error from OOProc workers. Rather, we just get one flat `RpcException` that encodes the error information as a single string value.  We could look into updating the Functions OOProc gRPC protocol to support nested exceptions (like I recently did for DTFx) but that would be a lot of work. Unless Functions could better support nested exceptions, nested activity exceptions can't really be supported.

Note that we __do__ support nested exceptions in orchestrator functions. This is because we control the full payload for errors. The limitation is only for activity functions.